src/test/tree/StrategyManangerUnit.tree

```diff
@@ -0,0 +1,89 @@
+StrategyManagerUnit.t.sol
+├── initialize
+│   └── when initialize called again
+│       └── it should revert
+├── depositIntoStrategy()
+│   ├── when deposit successfully
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposit successfully twice
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when safeTransferFrom() function reverts
+│   │   └── it should revert
+│   ├── when IERC20 token does not exist
+│   │   └── it should revert
+│   ├── when strategy.deposit() function reverts
+│   │   └── it should revert
+│   ├── when strategy does not exist
+│   │   └── it should revert
+│   ├── when strategy is not whitelisted
+│   │   └── it should revert
+│   └── _addShares()
+│       ├── when shares is 0
+│       │   └── it should revert
+│       └── when deposit would exceed max array length
+│           └── it should revert
+├── depositIntoStrategyWithSignature()
+│   ├── when deposit with signature successfully
+│   │   └── staker nonce should increase and shares should increase
+│   ├── when deposit with signature successfully and replay same deposit signature
+│   │   └── it should revert the second time
+│   ├── when deposit with signature with EIP1271 compliant contract signature
+│   │   └── it should deposit successfully with shares and nonce increase
+│   ├── when deposit with signature with EIP1271 compliant contract but with bad signature
+│   │   └── it should revert
+│   ├── when deposit with signature with non-compliant EIP1271 contract
+│   │   └── it should revert
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when signature expired
+│   │   └── it should revert
+│   └── when bad signature
+│       └── it should revert
+├── removeShares()
+│   ├── when not called by DelegationManager
+│   │   └── it should revert
+│   ├── when share amount is too high, higher than deposited amount
+│   │   └── it should revert
+│   ├── when deposit single strategy and remove all shares
+│   │   └── it should remove all shares and decrease stakerStrategyListLength (set to 0)
+│   ├── when deposited 3 strategies and randomly select one to remove all shares
+│   │   └── it should remove all shares and decrease stakerStrategyListLength (set to 2)
+│   └── when deposited 3 strategies and removeShares() from all 3 strategies with fuzzed inputs
+│       └── for each strategy, decrease shares properly but if sharesAfter is 0, then should decrement stakerStrategyListLength
```

Comment:
this is definitely describing tests rather than logical branches. the important branch is something like 'given the share amount is equal to the deposited amount'

Reply:
Refactored this to describe logical branches now